packages/next-swc/crates/next-core/src/app_render/next_server_component_transition.rs

```diff
@@ -59,10 +62,15 @@ impl Transition for NextServerComponentTransition {
             bail!("Not an ecmascript module");
         };
 
-        Ok(WithClientChunksAsset {
-            asset,
-            // next.js code already adds _next prefix
-            server_root: self.server_root.join("_next"),
+        Ok(WithChunkingContextScopeAsset {
+            asset: WithClientChunksAsset {
+                asset,
+                // next.js code already adds _next prefix
+                server_root: self.server_root.join("_next"),
+            }
+            .cell()
+            .into(),
+            layer: "rsc".to_string(),
```

Comment:
What does this fix?

Reply:
It puts these files in the correct layer. This fixes problems when using the same files in server components and in SSR. They would get the same chunk item ids and that leads to incorrect imports at runtime